src/FritzBox/Converter.php

```diff
@@ -8,35 +8,50 @@
 class Converter
 {
     private $config;
+    private $imagePath;
 
     public function __construct($config)
     {
-        $this->config = $config;
+        $this->config    = $config['conversions'];
+        $this->imagePath = $config['phonebook']['imagepath'] ?? NULL;
     }
 
     public function convert($card): SimpleXMLElement
     {
         $this->card = $card;
+        $foundEntry = false;
 
         // $contact = $xml->addChild('contact');
         $this->contact = new SimpleXMLElement('<contact />');
 
+        $this->contact->addChild('carddav_uid',$this->card->uid);    // reference for image upload
+        
         $this->addVip();
 
+        // add Person
         $person = $this->contact->addChild('person');
         $name = htmlspecialchars($this->getProperty('realName'));
         $person->addChild('realName', $name);
-        // $person->addChild('ImageURL');
-
-        $this->addPhone();
-        $this->addEmail();
-
-        $person = $this->contact->addChild('setup');
-
-        // print_r($this->contact);
-        // echo($this->contact->asXML().PHP_EOL);
+        
+        // add photo
+        if (isset($this->card->rawPhoto)) {
+            if (isset($this->imagePath)) {
+                $person->addChild('imageURL', $this->imagePath . $this->card->uid . '.jpg');
+            }
+        }
+        
+        $foundPhone = $this->addPhone();
+        
+        $foundEmail = $this->addEmail();
 
-        return $this->contact;
+        if ($foundEmail == true OR $foundPhone == true) {
+            return $this->contact;
+        }
+        else {                                                   // neither a phone number nor an email in this contact
+            $this->contact = new SimpleXMLElement('<void />');
```

Comment:
Warum braucht es das void element hier?

Reply:
Vorher landeten die Kontakte, die inhaltlich für FB wertlos waren/sind auch in der XML waren aber unkenntlich bzw. verstümmelt (Namen ohne Nummern) - in dieser Variante sieht man mit UID noch einen Rumpf (wer sich den XML-Export ansieht) hat aber keine Namen ohne Nummern/E-Mail im Telefonbuch.
Alternative: diese Kontakte rausfliegen lassen

Reply:
Warum nicht rausschmeissen statt kryptisches XML zu erzeugen?

Reply:
Na dann raus!

Reply:
Geht nicht - oder ich bin zu blöde!
da `public function convert($card): SimpleXMLElement` XML-Return erwartet habe ich jetzt versucht den return zu NULLen, wenn
`(!$foundEmail && !$foundPhone)`
mit `unset($this->contact)` dann mit `$this->contact = new SimpleXMLElement('')`,
fliegt aber alles auf die Bretter